spec/controllers/sign_up/recovery_codes_controller_spec.rb

```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+describe SignUp::RecoveryCodesController do
+  describe '#update' do
+    it 'redirects to the profile page' do
+      stub_sign_in
+      subject.current_user.recovery_code = 'foo'
+
+      patch :update
+
+      expect(response).to redirect_to profile_path
+    end
+  end
+end
```

Comment:
Can we please also add specs for the `#show` action, and the `confirm_has_not_already_viewed_recovery_code` before_action?